db/fetcher/encoded_doc.go

```diff
@@ -20,6 +20,18 @@ import (
 	"github.com/sourcenetwork/defradb/core"
 )
 
+type EncodedDocument interface {
```

Comment:
yes

Reply:
That's a pain - you sure there is no way to have this private?  I cant see why it would need to be public in Go.

Reply:
I think we might be misunderstanding each other.

If the interface is private, and located where it currently is, there appears to be nothing stopping the mock from implementing it in another package (where and how it currently is).

Only the code that touches both the mock and the interface need access to both types. And this way we dont need to change the production code exposure for the sake of the tests.

Reply:
Ahhh - I see the public `Fetcher` interface is previously returning a private type - that is quite bad, and reason enough to make the return type (interface or otherwise) public. Regardless of the desire to mock.

I'm happy with the current state :+1:

Reply:
https://github.com/sourcenetwork/defradb/issues/1552

Reply:
Trying to remember if there was a (bad) reason for this, or just accidentally.

I suspect the method returning the private type was itself private at some point, and either intentionally or accidentally exposed.